src/TimePicker/TimePicker.js

```diff
@@ -0,0 +1,242 @@
+// @flow weak
+
+import React, { Component, PropTypes } from 'react';
+// import { createStyleSheet } from 'jss-theme-reactor';
+// import classNames from 'classnames';
+import TextField, { TextFieldInput, TextFieldLabel } from '../TextField';
+import TimePickerDialog from './TimePickerDialog';
+import { formatTime } from '../utils/timeUtils';
+
+
+const emptyTime = new Date();
+emptyTime.setHours(0);
+emptyTime.setMinutes(0);
+emptyTime.setSeconds(0);
+emptyTime.setMilliseconds(0);
+
+/* export const styleSheet = createStyleSheet('TimePicker', (theme) => {
+  return {}
+});*/
+
+class TimePicker extends Component {
+  static propTypes = {
+    /**
+     * If true, automatically accept and close the picker on set minutes.
+     */
+    autoOk: PropTypes.bool,
+    /**
+     * Override the label of the 'Cancel' button.
+     */
+    cancelLabel: PropTypes.node,
+    /**
+     * The initial time value of the TimePicker.
+     */
+    defaultTime: PropTypes.object,
+    /**
+     * Override the inline-styles of TimePickerDialog's root element.
+     */
+    dialogStyle: PropTypes.object,
+    /**
+     * If true, the TimePicker is disabled.
+     */
+    disabled: PropTypes.bool,
+    /**
+     * Tells the component to display the picker in `ampm` (12hr) format or `24hr` format.
+     */
+    format: PropTypes.oneOf(['ampm', '24hr']),
+    /**
+     * text that should be shown as a label
+     */
+    hintText: PropTypes.string,
+    /**
+     * if true, the TimePicker as the landscape view
+     */
+    landscape: PropTypes.bool,
+    /**
+     * Override the label of the 'OK' button.
+     */
+    okLabel: PropTypes.node,
+    /**
+     * Callback function that is fired when the time value changes. The time value is passed in a Date Object.
+     * Since there is no particular event associated with the change the first argument will always be null
+     * and the second argument will be the new Date instance.
+     */
+    onChange: PropTypes.func,
+    /**
+     * Callback function fired when the TimePicker `TextField` gains focus.
+     */
+    onFocus: PropTypes.func,
+    /**
+     * Callback function fired when the TimePicker is tapped or clicked.
+     */
+    onTouchTap: PropTypes.func,
+    /**
+     * If true, uses ("noon" / "midnight") instead of ("12 a.m." / "12 p.m.").
+     *
+     * It's technically more correct to refer to "12 noon" and "12 midnight"
+     * rather than "12 a.m." and "12 p.m."
+     * and it avoids confusion between different locales.
+     * By default (for compatibility reasons) TimePicker uses
+     * ("12 a.m." / "12 p.m.").
+     */
+    pedantic: PropTypes.bool,
+    /**
+     * Override the inline-styles of the root element.
+     */
+    style: PropTypes.object,
+    /**
+     * Override the inline-styles of TimePicker's TextField element.
+     */
+    textFieldStyle: PropTypes.object,
```

Comment:
Any deep style properties can be removed now as users would no longer need to add `!important` on their CSS rules.